ceph/rados/core_workflows.py

```diff
@@ -1746,6 +1751,292 @@ def get_daemon_status(self, daemon_type, daemon_id) -> tuple:
+            pre_crash_report : output of "ceph crash ls" before the tests/ operations
+        Returns:
+            Pass -> True, fail -> False
+        """
+        # Load JSON data
+        orch_ps_data = pre_test_orch_ps
+        cmd = "ceph orch ps"
+        orch_ps_new_data = self.run_ceph_command(cmd=cmd)
+
+        # Function to collect daemon_name and hostname by daemon_type
+        def collect_daemon_info(data):
+            daemon_info = {}
+            for entry in data:
+                daemon_type = entry["daemon_type"]
+                daemon_name = entry["daemon_name"]
+                hostname = entry["hostname"]
+                if daemon_type not in daemon_info:
+                    daemon_info[daemon_type] = {}
+                daemon_info[daemon_type][daemon_name] = hostname
+            return daemon_info
+
+        # Collect daemon information from both JSON structures
+        orch_ps_daemons = collect_daemon_info(orch_ps_data)
+        orch_ps_new_daemons = collect_daemon_info(orch_ps_new_data)
+
+        # Compare daemons and print the results
+        all_match = True
+        for daemon_type, daemons in orch_ps_daemons.items():
+            if daemon_type in orch_ps_new_daemons:
+                for daemon_name, hostname in daemons.items():
+                    if daemon_name in orch_ps_new_daemons[daemon_type]:
+                        new_hostname = orch_ps_new_daemons[daemon_type][daemon_name]
+                        is_same = hostname == new_hostname
+                        if not is_same:
+                            all_match = False
+                            log.error(f"Daemon Type: {daemon_type}")
+                            log.error(f"  Daemon Name: {daemon_name}")
+                            log.error(f"    Old Hostname: {hostname}")
+                            log.error(f"    New Hostname: {new_hostname}")
+                            log.error(f"    Same Hostname: {is_same}")
+                    else:
+                        all_match = False
+                        log.error(f"Daemon Type: {daemon_type}")
+                        log.error(f"  Daemon Name: {daemon_name}")
+                        log.error(f"    Old Hostname: {hostname}")
+                        log.error("    New Hostname: Not found")
+                        log.error("    Same Hostname: False")
+            else:
+                all_match = False
+                log.error(f"Daemon Type: {daemon_type} not found post tests")
+
+        # Check for missing daemons in orch_ps_new
+        for daemon_type, daemons in orch_ps_daemons.items():
+            if daemon_type not in orch_ps_new_daemons:
+                all_match = False
+                log.error(f"Daemon Type: {daemon_type} is missing post tests")
+            else:
+                for daemon_name in daemons:
+                    if daemon_name not in orch_ps_new_daemons[daemon_type]:
```

Comment:
We cannot perform this check on the tests where we are deleting and adding OSD.

Reply:
Two ways we can go here:
1. Update the method implementation to pick checks based on flags sent. In the case you suggested, we can implement a flag, based on which, that particular check would not be run.
2. We will use the method call only post tests that we know would not change the placement of daemons on the cluster.